internal/newstate/statuses.go

```diff
@@ -0,0 +1,67 @@
+package newstate
+
+import "k8s.io/apimachinery/pkg/types"
+
+// Statuses holds the status-related information about Gateway API resources.
+// It is assumed that only a singe Gateway resource is used.
+type Statuses struct {
+	// the key is the name of a listener in the Gateway resource.
+	ListenerStatuses  map[string]ListenerStatus
```

Comment:
What do you think about making all of the maps for statuses their own types? It may help with readability.